docs/httpproxy.md

```diff
@@ -0,0 +1,833 @@
+  name: basic
+spec:
+  virtualhost:
+    fqdn: foo-basic.bar.com
+  routes:
+    - conditions:
+        prefix: /
+      services:
+        - name: s1
+          port: 80
+```
+
+**Lines 1-4**: As with all other Kubernetes objects, an HTTPProxy needs apiVersion, kind, and metadata fields. Note that the HTTPProxy API is currently considered beta.
+
+**Line 6-7**: The presence of the `virtualhost` field indicates that this is a root HTTPProxy that is the top level entry point for this domain.
+The `fqdn` field specifies the fully qualified domain name that will be used to match against `Host:` HTTP headers.
+
+**Lines 8-9**: Each HTTPProxy must have one or more `routes`, each of which must have a condition to match against (e.g. `prefix: /`) and then one or more `services` which will handle the HTTP traffic.
+
+**Lines 10-12**: The `services` field is an array of named Service & Port combinations that will be used for this HTTPProxy path.
+HTTP traffic will be sent directly to the Endpoints corresponding to the Service.
+
+## Interacting with HTTPProxies
+
+As with all Kubernetes objects, you can use `kubectl` to create, list, describe, edit, and delete HTTPProxy CRDs.
+
+Creating an HTTPProxy:
+
+```bash
+$ kubectl create -f basic.httpproxy.yaml
+httpproxy "basic" created
+```
+
+Listing HTTPProxies:
+
+```bash
+$ kubectl get httpproxy
+NAME      AGE
+basic     24s
+```
+
+Describing HTTPProxy:
+
+```bash
+$ kubectl describe httpprox basic
+Name:         basic
+Namespace:    default
+Labels:       <none>
+API Version:  projectcontour.io/v1alpha1
+Kind:         HTTPProxy
+Metadata:
+  Cluster Name:
+  Creation Timestamp:  2019-07-05T19:26:54Z
+  Resource Version:    19373717
+  Self Link:           /apis/projectcontour.io/v1alpha1/namespaces/default/httpproxy/basic
+  UID:                 6036a9d7-8089-11e8-ab00-f80f4182762e
+Spec:
+  Routes:
+    Conditions:
+      Prefix: /
+    Services:
+      Name:  s1
+      Port:  80
+  Virtualhost:
+    Fqdn:  foo-basic.bar.com
+Events:    <none>
+```
+
+Deleting HTTPProxies:
+
+```bash
+$ kubectl delete httpproxy basic
+httpproxy "basic" deleted
+```
+
+## HTTPProxy API Specification
+
+TODO(dfc) create example directory
+
+There are a number of [working examples](/examples/example-workload/httpproxy) of HTTPProxy objects in the `examples/example-workload` directory.
+
+We will use these examples as a mechanism to describe HTTPProxy API functionality.
+
+### Virtual Host Configuration
+
+#### Fully Qualified Domain Name
+
+Similar to Ingress, HTTPProxy support name-based virtual hosting.
+Name-based virtual hosts use multiple host names with the same IP address.
+
+```
+foo.bar.com --|                 |-> foo.bar.com s1:80
+              | 178.91.123.132  |
+bar.foo.com --|                 |-> bar.foo.com s2:80
+```
+
+Unlike Ingress, HTTPProxy only support a single root domain per HTTPProxy object.
+As an example, this Ingress object:
+
+```yaml
+# ingress-name.yaml
+apiVersion: extensions/v1beta1
```

Comment:
Should we switch the examples to use the new group? (e.g. `networking.k8s.io/v1beta1`)